Automattic/jetpack
# composer-plugin: Generate i18n mapping for Assets

#### Changes proposed in this Pull Request:

Collect the `.extra.textdomain` fields from all the installed
jetpack-library packages and write the data into
`jetpack_vendor/i18n-map.php`.

Eventually we'll have each plugin pass that file to the
`Assets::alias_textdomains_from_file()` method being added to the Assets
package in #22083 to register the mappings for each Composer package.

#### Jetpack product discussion


#21690

#### Does this pull request change what data or activity we track or use?


No

#### Testing instructions:






* Check that the Build job's assets are including a `jetpack_vendor/i18n-map.php` file in the Jetpack, Boost, and Backup plugins, with a format matching that required by `Assets::alias_textdomains_from_file()` in #22083. The 'packages' list should be empty at this time.
* Edit some of the packages to specify `.extra.textdomain` in their composer.json files, then do `tools/composer-update-monorepo.sh projects/plugins/jetpack` and see if `projects/plugins/jetpack/jetpack_vendor/i18n-map.php` has the correct textdomains and versions in the 'packages' list.
